src/runtime/sgx/trusted/ecall_registry.cc

```diff
@@ -0,0 +1,109 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file ecall_registry.cc
+ * \brief The global registry of packed functions available via ecall_packed_func.
+ */
+#include <dmlc/logging.h>
+#include <tvm/runtime/registry.h>
+
+namespace tvm {
+namespace runtime {
+namespace sgx {
+
+class ECallRegistry: public Registry {
+ public:
+  struct Manager {
+    std::vector<ECallRegistry> exports;
```

Comment:
I feel this registry implementation is overly complicated. Maybe we can simply do things like.
```c++
ocall_register_func(fid, const char* name);

in each register call. We do need to have a convention that malloc the name and have the register_func free it later
```